client/pages/product-details.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import Container from 'react-bootstrap/Container';
+import parseRoute from '../lib/parse-route';
+import Row from 'react-bootstrap/Row';
+import ListGroup from 'react-bootstrap/ListGroup';
+import Accordion from 'react-bootstrap/Accordion';
+
+export default class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product: {},
+      route: parseRoute(window.location.hash),
+      loading: false
+    };
+    this.sizes = this.sizes.bind(this);
+  }
+
+  componentDidMount() {
+    const { params } = this.state.route;
+    const paramObject = Object.fromEntries(params);
+    const productId = paramObject.product;
+    fetch(`api/products/${productId}`)
+      .then(res => res.json())
+      .then(product => this.setState({ product, loading: true }))
+      .catch(err => console.error(err));
+    if (this.state.product === {}) {
+      this.setState({ loading: false });
+    }
```

Comment:
Objects never equal each other in JavaScript, because they are different memory addresses.